composeml/label_times/object.py

```diff
@@ -0,0 +1,491 @@
+            right (bool) : Indicates whether bins includes the rightmost edge or not. Does not apply to quantile-based bins.
+
+        Returns:
+            LabelTimes : Instance of labels.
+
+        Examples:
+            These are the target values for the examples.
+
+            >>> data = [226.93, 47.95, 283.46, 31.54]
+            >>> lt = LabelTimes({'target': data})
+            >>> lt
+               target
+            0  226.93
+            1   47.95
+            2  283.46
+            3   31.54
+
+            Bin values using equal-widths.
+
+            >>> lt.bin(2)
+                        target
+            0  (157.5, 283.46]
+            1  (31.288, 157.5]
+            2  (157.5, 283.46]
+            3  (31.288, 157.5]
+
+            Bin values using custom-widths.
+
+            >>> lt.bin([0, 200, 400])
+                   target
+            0  (200, 400]
+            1    (0, 200]
+            2  (200, 400]
+            3    (0, 200]
+
+            Bin values using quartiles.
+
+            >>> lt.bin(4, quantiles=True)
+                                     target
+            0             (137.44, 241.062]
+            1              (43.848, 137.44]
+            2             (241.062, 283.46]
+            3  (31.538999999999998, 43.848]
+
+            Assign labels to bins.
+
+            >>> lt.bin(2, labels=['low', 'high'])
+              target
+            0   high
+            1    low
+            2   high
+            3    low
+        """  # noqa
+        label_times = self.copy()
+        values = label_times[self.label_name].values
+
+        if quantiles:
+            label_times[self.label_name] = pd.qcut(values, q=bins, labels=labels)
+
+        else:
+            label_times[self.label_name] = pd.cut(values, bins=bins, labels=labels, right=right)
+
+        transform = {
+            'transform': 'bin',
+            'bins': bins,
+            'quantiles': quantiles,
+            'labels': labels,
+            'right': right,
+        }
+
+        label_times.transforms.append(transform)
+        label_times.label_type = 'discrete'
+        return label_times
+
+    def _sample(self, key, value, settings, random_state=None, replace=False):
+        """Returns a random sample of labels.
+
+        Args:
+            key (str) : Determines the sampling method. Can either be 'n' or 'frac'.
+            value (int or float) : Quantity to sample.
+            settings (dict) : Transform settings used for sampling.
+            random_state (int) : Seed for the random number generator.
+            replace (bool) : Sample with or without replacement. Default value is False.
+
+        Returns:
+            LabelTimes : Random sample of labels.
+        """
+        sample = super().sample(random_state=random_state, replace=replace, **{key: value})
+        recursive = getattr(self, '_recursive', False)
+
+        if not recursive:
```

Comment:
what if, instead of using this `recursive` flag to determine if we need to add the transform settings, we handled appending to the the transform settings in the `sample` method, after calling `_sample` or `_sample_per_label`.

Reply:
Great idea, that works as well. Thanks!